### DISCUSSION TITLE
Introduce new option in `migrations:create` command to give a name for migrations files

### DISCUSSION THREAD

**Discussion by AbdlrahmanSaberAbdo** [mikro-orm/mikro-orm#3987] [Ideas]
I'd like to add a new option in `migrations:create` command to give the ability to developers to give specific names for `migrations` files, I came from `Laravel` and  `Ruby on rails` world and I had the ability to give the migration file a name.

## Advantages 
- Better Organization: Naming migration files in a meaningful way helps keep the migrations organized and easy to understand. It also makes it easier to search for a specific migration when needed.

- Improved Documentation: When migration files are named in a descriptive way, it becomes easier to understand the changes being made to the database, even if you don't have access to the code. This can help with future maintenance and upgrades.

- Easier Debugging: In the event of a problem, being able to easily locate the migration that caused the issue can save time and simplify the debugging process.

I saw @B4nan suggestions that we can update the configuration, but I'm not sure if we can specific name for each file, also will be much easier if we just put the file name when write `migrations:create` command

## Examples
- Let's say I create a table called `orders` so the migration file will be called `timestamp_create_orders_table`.
- At some point, I needed to add new columns or update existing columns so the new migration file will be called `timestamp_add_rate_column` or `timestamp_update_order_id_column_to_id`.

I think this will be helpful, what do you think guys?

**Comment by B4nan** [comment]
> I saw @B4nan suggestions that we can update the configuration, but I'm not sure if we can specific name for each file, also will be much easier if we just put the file name when write migrations:create command

You can configure the pattern (`glob` option) for migration file matching, and you can as well override the default `fileName` via callback. Since v5, the default `glob` pattern is generous - `!(*.d).{js,ts}` - it accepts any JS/TS file, so you shouldn't need to care about that. The only real requirement is that your migration file names are sequential so we can sort them.

So what you could do right now is to override the `fileName` callback to have the file name start with the timestamp, and then rename it to whatever you want. The migrator should find the file just fine, as long as the naming is sequential, things should just work.

We could have a guide on this for sure, we could add an option to the CLI too, but it feels a bit dangerous, given the default naming strategy wouldn't work with that (as timestamp is at the end).

**Reply by AbdlrahmanSaberAbdo** [reply]
> > I saw @B4nan suggestions that we can update the configuration, but I'm not sure if we can specific name for each file, also will be much easier if we just put the file name when write migrations:create command
> 
> You can configure the pattern (`glob` option) for migration file matching, and you can as well override the default `fileName` via callback. Since v5, the default `glob` pattern is generous - `!(*.d).{js,ts}` - it accepts any JS/TS file, so you shouldn't need to care about that. The only real requirement is that your migration file names are sequential so we can sort them.
> 
> So what you could do right now is to override the `fileName` callback to have the file name start with the timestamp, and then rename it to whatever you want. The migrator should find the file just fine, as long as the naming is sequential, things should just work.
> 
> We could have a guide on this for sure, we could add an option to the CLI too, but it feels a bit dangerous, given the default naming strategy wouldn't work with that (as timestamp is at the end).

Sounds good, will try to override the `filename` callback and see,  and will do a guide on this if it works perfectly. 
regarding CLI, I didn't get you well, but the timestamp will be at the start not at the end so the name will be sequential.

**Comment by srmagura** [comment]
It would be very useful to be able to name migrations via a CLI arg! E.g.

```shell
yarn mikro-orm migration:create --name SamTest
```

would create a file called `Migration20230324194131_SamTest.ts`, or something similar to that.

**Reply by srmagura** [reply]
Let me know if I can submit a PR that implements this.

**Reply by AbdlrahmanSaberAbdo** [reply]
Hi @srmagura, sorry for the delay, I just created a PR today for it. https://github.com/mikro-orm/mikro-orm/pull/4250, hope to be merged soon
